### PR TITLE
Add information to SAML parser exceptions

### LINKED ISSUE #29698: Improve SAML2 Metadata Validation Exception messages
### Description

When Keycloak encounters an error while parsing of an entity descriptor metadata document, the exception message doesn't give much information about where in the document the parsing error occurred.

### Discussion

https://github.com/keycloak/keycloak/discussions/29694

### Motivation

it would be easier to understand which part of the metadata document is invalid if the exception message contains more information, such as:
* expected and found namespace
* line numner and column

### Details

This is an example of how a parsing exception is currently logged:
`PL00066: Parser : Expected start tag:Attribute.  Found <Attribute>`

Proposal for the extended parsing exception with more information:
`PL00066: Parser : Expected start tag: (urn:oasis:names:tc:SAML:2.0:metadata)Attribute. Found (urn:oasis:names:tc:SAML:metadata:attribute)Attribute. At line 41, column 154`

### PR DESCRIPTION
**Pull request by esagalara**
Include namespaces and location of expected/found elements

Closes #29698

### CODE REVIEW

**Comment by ahus1** [comment]
@rmartinc / @mposolda - as this PR touches SAML, maybe you can review it when you have the time. Thanks!

**Review by rmartinc — CHANGES_REQUESTED** [review]
Thanks @esagalara! Just commenting about replacing the method instead of adding a new one, and this way chage all the usages (it seems there are two). WDYT?

File: saml-core/src/main/java/org/keycloak/saml/common/DefaultPicketLinkLogger.java [hunk 1]
```diff
@@ -451,6 +451,12 @@ public RuntimeException parserExpectedTag(String tag, String foundElementTag) {
         return new RuntimeException(ErrorCodes.EXPECTED_TAG + tag + ".  Found <" + foundElementTag + ">");
     }
 
+    @Override
+    public RuntimeException parserExpectedTag(String tag, String foundElementTag, Integer line, Integer column) {
+        return new RuntimeException(ErrorCodes.EXPECTED_TAG + " " + tag + ".  Found " + foundElementTag +
+                ".\nAt line " + line.toString() + ", column " + column);
```
[the remark this answers is not in the document]
Comment (esagalara) [reply saml-core/src/main/java/org/keycloak/saml/common/DefaultPicketLinkLogger.java:457]: Sure, that sounds good.

[the remark this answers is not in the document]
Comment (esagalara) [reply saml-core/src/main/java/org/keycloak/saml/common/DefaultPicketLinkLogger.java]: I have removed the previous version of the `parserExpectedTag` method now.

**Review by rmartinc — APPROVED** [review]
Thanks @esagalara!

**Review by mposolda — APPROVED** [review]
@rmartinc Thanks for the review!

**Pull request merged**